src/messages/Notification.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  constant,
+  Decoder,
+  either,
+  map,
+  mixed,
+  object,
+  optional,
+  string
+} from 'decoders';
+import { LabeledMsg } from './LabeledMsg';
+
+/**
+ * A toast configuration.
+ */
+export interface Notification {
+  /** The title of the notification */
+  title?: string;
+  /** The notification message */
+  message: string;
+  /** Additional host-specific options such as severity */
+  custom: any;
+}
+
+/**
+ * A message used to request toasts to display
+ * in the host application.
+ * @external
+ */
+export interface LabeledNotification extends LabeledMsg {
+  /** Message identifier */
+  msgType: 'notifyRequest';
+  /** Toast details */
+  msg: Notification;
+}
+
+/**
+ * Helper function to convert old message types to the new type
+ * @external
+ */
+function alwaysMsgType(msgType: 'string'): 'notifyRequest' {
+  return 'notifyRequest';
+}
+
+/** @external */
+const toastTypeDecoder: Decoder<'notifyRequest'> = map(
+  constant<'toastRequest'>('toastRequest'),
+  alwaysMsgType
+);
+
+/** @external */
+const decoder: Decoder<LabeledNotification> = object({
+  msgType: either(constant<'notifyRequest'>('notifyRequest'), toastTypeDecoder),
```

Comment:
It's a bit hard to see what changed here with the file being renamed, but the `either` here combined with `toastTypeDecoder` and `alwaysMsgType` treats messages with `msgType: toastRequest` as valid and decodes them to a value with `msgType: notifyRequest`. 

This is the other really nice thing about decoders. In addition to detecting bad data at the edges of your system, they provide a well-defined location for handling compatibility when an API changes.